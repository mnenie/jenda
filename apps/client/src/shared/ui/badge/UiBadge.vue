<script setup lang="ts">
import { type BadgeVariants, badgeVariants } from '.'
import type { HTMLAttributes } from 'vue'
import { cn } from '@/shared/libs/shadcn/utils'

const { variant = 'default', class: badgeClass } = defineProps<{
  variant?: BadgeVariants['variant']
  class?: HTMLAttributes['class']
  closable?: boolean
  closeIcon?: HTMLAttributes['class']
}>()

const emit = defineEmits<{
  (e: 'close'): void
}>()
</script>

<template>
  <div :class="cn(badgeVariants({ variant }), badgeClass)">
    <slot />
    <div
      v-if="closable"
      i-lucide-x
      class="ml-0.5 flex items-center w-3.5 h-3.5 text-neutral-500 dark:text-neutral-400 cursor-pointer"
      :style="{ color: closeIcon }"
      @click="emit('close')"
    />
  </div>
</template>
