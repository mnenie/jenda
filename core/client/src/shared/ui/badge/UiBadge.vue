<script setup lang="ts">
import { type BadgeVariants, badgeVariants } from '.'
import type { HTMLAttributes } from 'vue'
import { cn } from '@/shared/libs/shadcn/utils'

const { variant = 'default', class: badgeClass } = defineProps<{
  variant?: BadgeVariants['variant']
  class?: HTMLAttributes['class']
}>()
</script>

<template>
  <div :class="cn(badgeVariants({ variant }), badgeClass)">
    <slot />
  </div>
</template>
