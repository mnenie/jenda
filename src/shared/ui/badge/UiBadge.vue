<script setup lang="ts">
import type { BadgeType } from './types';

withDefaults(
  defineProps<{
    variant?: BadgeType;
  }>(),
  {
    variant: 'default'
  }
);
</script>

<template>
  <div class="text-xs" :class="[$style.badge, $style[variant]]">
    <slot />
  </div>
</template>

<style module lang="scss">
.badge {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  white-space: nowrap;
  border-radius: 6px;
  font-weight: 500 !important;
  padding: 1px 10px;
}

.default {
  background-color: var(--zinc-900);
  color: var(--zinc-50);
}

.secondary {
  background-color: rgba(228, 228, 231, 0.8);
  color: var(--zinc-950);
}

.outline {
  border: 1px solid var(--zinc-200);
  color: var(--zinc-900);
  background-color: white;
}

:global(html.dark) {
  .default {
    background-color: var(--zinc-100);
    color: var(--zinc-900);
  }

  .secondary {
    background-color: rgba(var(--zinc-rgb-600), 0.5);
    color: var(--zinc-200);
  }
}
</style>
