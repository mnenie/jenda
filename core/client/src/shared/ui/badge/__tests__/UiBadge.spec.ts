import type { BadgeType } from "../types";
import { shallowMount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import UiBadge from "../UiBadge.vue";

describe("uiBadge", () => {
  const wrapper = shallowMount(UiBadge, {
    props: {
      variant: "default",
    },
  });

  it("should render correctly", () => {
    expect(wrapper.html());
  });

  it("should have content at slot", () => {
    const _w = shallowMount(UiBadge, {
      slots: {
        default: "badge",
      },
    });
    expect(_w.html()).toMatchSnapshot();
  });

  it("should apply correct styles for variants", () => {
    const variants = ["default", "secondary", "outline"] as BadgeType[];
    variants.forEach((variant) => {
      const wrapper = shallowMount(UiBadge, {
        props: {
          variant,
        },
      });
      const badge = wrapper.find("div");
      expect(badge.attributes("class")).toContain(`${variant}`);
    });
  });
});
