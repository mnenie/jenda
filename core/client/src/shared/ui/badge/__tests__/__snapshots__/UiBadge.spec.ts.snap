// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UiBadge > should have content at slot 1`] = `"<div class="text-xs badge default">badge</div>"`;

exports[`uiBadge > should have content at slot 1`] = `"<div class="text-xs badge default">badge</div>"`;
